src/test/scala/io/delta/sql/parser/DeltaSqlParserSuite.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2019 Databricks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.sql.parser
+
+import io.delta.tables.execution.VacuumTableCommand
+
+import org.apache.spark.SparkFunSuite
+import org.apache.spark.sql.catalyst.TableIdentifier
+
+class DeltaSqlParserSuite extends SparkFunSuite {
+
+  test("isValidDecimal should recognize a table identifier and not treat them as a decimal") {
+    // Setting `delegate` to `null` is fine. The following tests don't need to touch `delegate`.
+    val parser = new DeltaSqlParser(null)
+    assert(parser.parsePlan("vacuum 123_") ===
```

Comment:
@mukulmurthy I ported tests from https://github.com/apache/spark/commit/a6b8182006d0c3dda67c06861067ca78383ecf1b#diff-c1e3256a0af258699e47c4d7f043b383R523 and also verified this test would fail if I broke `isValidDecimal`.